sherpa/astro/flux.py

```diff
@@ -584,22 +578,16 @@ def print_sample_result(title, arg):
             session.set_full_model(id, orig_model)
             mystat.append(session.calc_stat(id))
             #####################################
-        oflxiflx = [oflx, iflx]
 
-        myconfidence = (1.0 - confidence / 100.0) / 2.0
+        hwidth = confidence / 2
         result = []
+        for x in [oflx, iflx]:
+            result.append(numpy.percentile(x, [50, 50 + hwidth, 50 - hwidth]))
 
-        for x in oflxiflx:
-            sf = numpy.sort(x)
-            median = numpy.median(sf)
-            upconfidence_index = int((1.0 - myconfidence) * size - 1)
-            loconfidence_index = int(myconfidence * size - 1)
-            upconfidence = sf[upconfidence_index]
-            loconfidence = sf[loconfidence_index]
-            result.append(numpy.array([median, upconfidence, loconfidence]))
-
-        print_sample_result('original model flux', result[0])
-        print_sample_result('model component flux', result[1])
+        for lbl, arg in zip(['original model', 'model component'], result):
+            med, usig, lsig = arg
+            msg = '{} flux = {:g}, + {:g}, - {:g}'.format(lbl, med, usig - med, med - lsig)
+            print(msg)
```

Comment:
This is tricky. I would really want this to be a `logger.INFO` message. I thought we want to move away form print statements in general, because users can't filter or switch them off if they run this in a script. On the other hand, this is inside a block that resets the log level so we want simply do that. 
The code above resets the base sherpa logger to ERROR, so even a new logging instance just for this function would not show INFO messages. Maybe we can make the line above that silences logging of other functions more specific and then create a new logger here that emits the output?

Reply:
The problem with breaking up #754 into manageable chunks is that some of the follow-on clean-up code context is missed. There are planned logging changes coming - e.g. 275ea0fd6aeb002fd3ed8178fa9dcf531b85495f and then d40677607ad7d95f7cd385f23d2061e7b67c6115 and then actually adding a test at db3243488e9a15105a3dbd6a58bff84793899b7b - but they are not easy to separate out from some of the other changes I made in that PR.

So I hope to be able to assuage your concerns with Yet Another PR soon ;-)